src/components/CreateDeliveryModal.tsx

```diff
@@ -0,0 +1,87 @@
+import React, { useState } from "react";
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import { addDelivery } from "../fake-db";
+
+const CreateDeliveryModal = (props: {
+  x: boolean;
+  setX: (x: boolean) => void;
+}) => {
+  const { x, setX } = props;
+  const queryClient = useQueryClient();
+  const [isLoading, setIsLoading] = useState(false);
+
+  const { mutate } = useMutation({
+    mutationFn: (opts: {
+      name: string;
+      status: string;
+      deliveryDate: Date;
+    }) => addDelivery(opts.name, opts.status, opts.deliveryDate as any as string),
```

Comment:
i would change this to:
```suggestion
mutationFn: addDelivery
```

and go back to `addDelivery` and update it to take in an object. this way you don't have to redeclare the types (and mismatches like the one you have here will be less likely to happen) and it works very nicely with `useMutation`